Implemented dark mode feature

yogaprasadk opened the pull request:
This PR introduces a dark mode toggle feature for the sorting visualizer, allowing users to switch between light and dark themes with ease. The feature has been tested, and it persists even after page reloads.

Changes Made:
    Added a dark mode toggle button to the UI.
    Implemented the logic for switching between light and dark modes using JavaScript and CSS.
    Applied smooth transition effects for switching themes.
    Saved user preferences using localStorage to retain the theme on page reload.

 Tags:
    #apertre2.0
    #Difficulty: Medium

<h3>Light Mode</h3>
<img src="https://github.com/user-attachments/assets/a0c61e4f-21d0-48d5-a128-ea0b3d80efe0"/>

<h3>Dark Mode</h3>
<img src="https://github.com/user-attachments/assets/8852ca8a-de1b-43f4-b7f7-d637dde9e014"/>

Ayushjhawar8 commented:
Hey @yogaprasadk tysm for pr, can you make the header the same green shade as light mode

yogaprasadk commented:
ok bro

yogaprasadk commented:
between light and dark themes with ease. The feature has been tested, and it persists even after page reloads.

Changes Made:
    Added a dark mode toggle button to the UI.
    Implemented the logic for switching between light and dark modes using JavaScript and CSS.
    Applied smooth transition effects for switching themes.
    Saved user preferences using localStorage to retain the theme on page reload.

 Tags:
    #apertre2.0
    #Difficulty: Medium

<h3>Light Mode</h3>
<img src="https://github.com/user-attachments/assets/0d4bece7-e589-48cb-ac25-e947e7e4768e"/>

<h3>Darkmode</h3>
<img src="https://github.com/user-attachments/assets/8b18b1cb-1f59-4ec6-87ba-ab6524ac418b"/>

Ayushjhawar8 commented:
LGTM!
